### DISCUSSION TITLE
[mlp] Is it possible to deploy only specific microservices?

### DISCUSSION THREAD

**Discussion by frengooo** [mia-platform/community#401] [Q&A] [answered]
### What product are you having troubles with?

Console

### What Console version are you using?

v12.0.1

### Description

Hi guys,
I need to trigger pipelines to deploy only one specific microservice.
The flow is the following one:
* commit on a microservice;
* build pipeline starts;
* build and push done;
* deploy pipeline trigger, deploying the built microservice only.

Is this possible?

**Comment by davidebianchi** [comment, marked as answer]
Hi @silversoul93, I think it's needed some more context to help you.
If you build a latest tag of a specific service, and the service is running with the same tag, you can delete the specific pod using miactl. I start from the assumption that it is released without any changes to the k8s configuration.

The imagePullPolicy should be set from the Console to `Always` if you don't overwrite it.

What do you think about?

**Reply by frengooo** [reply]
Hi @davidebianchi,
thanks for your reply. I forgot the write the assumption, but it's right, we will release without modifying the k8s manifest.

With the `imagePullPolicy: Always` the `delete pod` workaround should work.

I think there should be better an explicit command to do that, regardless of the `imagePullPolicy`, but we can use the `delete` command for now.
Do you think it would be a useful feature?

**Reply by davidebianchi** [reply]
Maybe a rollout command could be interesting. Anyway, the `imagePullPolicy` is a K8s configuration so it works when pod is restarted. The possibilities are to set it to `Always` (as it is at the moment), or to change the tag of the docker image (e.g. create a docker image per commit).

**Reply by frengooo** [reply]
> Maybe a rollout command could be interesting.

I agree, it should be very useful.


> The possibilities are to set it to Always (as it is at the moment), or to change the tag of the docker image (e.g. create a docker image per commit).

For now I think we are going to use the `delete pod` approach, it's not so comfortable to always change the image performing a commit.
Thanks.
